example/project_benchmark.exs

```diff
@@ -0,0 +1,85 @@
+defmodule Clickhouse.Benchmark.Performance do
+  use Benchee.Benchmark
+
+  # Since it's a module, any other constructs may be used,
+  # such as defs, defmacro, etc.
+  def map_fun(i) do
+    [i]
+  end
+
+  # Global benchmark setup. Optional.
+  #
+  # Called before any benchmark in the module.
+  #
+  # If defined, must return {:ok, state} for benchmarks to run
+  setup do
+    {:ok, nil}
```

Comment:
It's rather a matter of convention. `raise` means some serious runtime error that should have never happened if the code were write. 
Surely, we can, just `throw` things, but it will be more implicit I think: if you just look on the result type of the setup, you already know what it should return to work correctly. To tell programmers to just throw their error would be less obvious IMO than that.

Reply:
I just never quite understood it. In what scenario will my setup not work? Usually the code I call will already raise at some point. It might just be me, I've never seen `setup` implementation that returns anything but `{:ok, state}`

Reply:
Well, the code that returns ok/error pair usually should never throw (although might raise), at least with any other reason then MatchError, but that's on the caller side, so not function's problem, but programmer that's calling it :)

Surely it might be easier to write happy-path-code all the way, just a matter of taste I guess.

Share your final thoughts on this :)

Reply:
I'm fine with it for ExUnits sake :man_shrugging: